src/mocker.rs

```diff
@@ -0,0 +1,216 @@
+#[cfg(feature = "mocker")]
+use std::collections::HashMap;
+use std::net::{SocketAddr, TcpListener};
+use std::time::Duration;
+
+use crate::client::GrowthBookClient;
+use crate::error::GrowthbookError;
+use rand::Rng;
+use serde_json::{Value, json};
+use uuid::Uuid;
+use wiremock::matchers::{method, path};
+use wiremock::{Mock, MockServer, ResponseTemplate};
+
+#[cfg(feature = "mocker")]
+pub struct GrowthBookTest {
+    mock_server: MockServer,
+    client: GrowthBookClient,
+    sdk_key: Uuid,
+    features: HashMap<String, Value>,
+}
+
+#[cfg(feature = "mocker")]
+impl GrowthBookTest {
+    pub async fn new() -> Result<Self, GrowthbookError> {
+        let mock_server = Self::create_mock_server().await;
+        let sdk_key = Uuid::now_v7();
+        let features = HashMap::new();
+
+        // Configura o mock inicial
+        let body = json!({
+            "status": 200,
+            "features": features,
+            "dateUpdated": "2024-05-29T18:43:22.153Z"
+        });
+
+        Mock::given(method("GET"))
+            .and(path(format!("/api/features/{}", sdk_key)))
+            .respond_with(ResponseTemplate::new(200).set_body_string(body.to_string()))
+            .expect(1..) // accept many requests
+            .mount(&mock_server)
+            .await;
+
+        let client = GrowthBookClient::new(
+            &mock_server.uri(),
+            &sdk_key.to_string(),
+            Some(Duration::from_millis(100)), // short interval for tests and make the things "atomic"
+            None,
+        )
+        .await?;
+
+        // wait time for the first time to get the features
+        tokio::time::sleep(Duration::from_millis(150)).await;
```

Comment:
is this reliable? Should we be able to await or force the feature loading directly?